compiler/rustc_abi/src/lib.rs

```diff
@@ -1828,11 +1828,12 @@ where
 pub enum PointerKind {
     /// Shared reference. `frozen` indicates the absence of any `UnsafeCell`.
     SharedRef { frozen: bool },
-    /// Mutable reference. `unpin` indicates the absence of any pinned data.
-    MutableRef { unpin: bool },
-    /// Box. `unpin` indicates the absence of any pinned data. `global` indicates whether this box
-    /// uses the global allocator or a custom one.
-    Box { unpin: bool, global: bool },
+    /// Mutable reference. `unsafe_unpin` indicates the absence of any pinned
+    /// data (`UnsafePinned` or `PhantomPinned`).
+    MutableRef { unsafe_unpin: bool },
```

Comment:
This should keep the name `unpin` -- it refers to the trait, just like `frozen` above refers to `Freeze`.